tuxedo/pulse/15/gen2/default.nix

```diff
@@ -0,0 +1,22 @@
+{
+  lib,
+  pkgs,
+  ...
+}: {
+  imports = [
+    ../../../../common/cpu/amd
+    ../../../../common/gpu/amd
+    ../../../../common/pc/laptop/ssd
+  ];
+
+  hardware.amdgpu.loadInInitrd = lib.mkDefault false;
```

Comment:
I think this is more related to DRM-enabled displays. Are you sure it's actually going away by having this set? It could also be the case that the error is just printed later. You can check `dmesg` to see if this is the case.

Reply:
> Are you sure it's actually going away by having this set? It could also be the case that the error is just printed later. You can check dmesg to see if this is the case.

Indeed! The error message appears in both cases.

Declaring `hardware.amdgpu.loadInInitrd = false;` results in:

```bash
$ dmesg | grep amdgpu
[   21.286293] [drm] amdgpu kernel modesetting enabled.
[   21.289984] amdgpu: Virtual CRAT table created for CPU
[   21.289997] amdgpu: Topology: Add CPU node
[   21.290117] amdgpu 0000:05:00.0: enabling device (0006 -> 0007)
[   21.312042] amdgpu 0000:05:00.0: amdgpu: Fetched VBIOS from VFCT
[   21.312046] amdgpu: ATOM BIOS: 113-LUCIENNE-017
[   21.313847] amdgpu 0000:05:00.0: amdgpu: Trusted Memory Zone (TMZ) feature enabled
[   21.313858] amdgpu 0000:05:00.0: amdgpu: MODE2 reset
[   21.314118] amdgpu 0000:05:00.0: amdgpu: VRAM: 512M 0x000000F400000000 - 0x000000F41FFFFFFF (512M used)
[   21.314120] amdgpu 0000:05:00.0: amdgpu: GART: 1024M 0x0000000000000000 - 0x000000003FFFFFFF
[   21.314121] amdgpu 0000:05:00.0: amdgpu: AGP: 267419648M 0x000000F800000000 - 0x0000FFFFFFFFFFFF
[   21.314177] [drm] amdgpu: 512M of VRAM memory ready
[   21.314178] [drm] amdgpu: 15744M of GTT memory ready.
[   21.319187] amdgpu 0000:05:00.0: amdgpu: PSP runtime database doesn't exist
[   21.319191] amdgpu 0000:05:00.0: amdgpu: PSP runtime database doesn't exist
[   21.343293] amdgpu 0000:05:00.0: amdgpu: Will use PSP to load VCN firmware
[   22.143231] amdgpu 0000:05:00.0: amdgpu: RAS: optional ras ta ucode is not available
[   22.152586] amdgpu 0000:05:00.0: amdgpu: RAP: optional rap ta ucode is not available
[   22.157726] amdgpu 0000:05:00.0: amdgpu: Secure display: Generic Failure.
[   22.157735] amdgpu 0000:05:00.0: amdgpu: SECUREDISPLAY: query securedisplay TA failed. ret 0x0
[   22.160691] amdgpu 0000:05:00.0: amdgpu: SMU is initialized successfully!
[   22.193604] snd_hda_intel 0000:05:00.1: bound 0000:05:00.0 (ops amdgpu_dm_audio_component_bind_ops [amdgpu])
[   22.318529] kfd kfd: amdgpu: Allocated 3969056 bytes on gart
[   22.318593] amdgpu: sdma_bitmap: 3
[   22.318610] amdgpu: SRAT table not found
[   22.318610] amdgpu: Virtual CRAT table created for GPU
[   22.318949] amdgpu: Topology: Add dGPU node [0x164c:0x1002]
[   22.318953] kfd kfd: amdgpu: added device 1002:164c
[   22.319041] amdgpu 0000:05:00.0: amdgpu: SE 1, SH per SE 1, CU per SH 8, active_cu_number 8
[   22.319090] amdgpu 0000:05:00.0: amdgpu: ring gfx uses VM inv eng 0 on hub 0
[   22.319091] amdgpu 0000:05:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
[   22.319092] amdgpu 0000:05:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
[   22.319093] amdgpu 0000:05:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 5 on hub 0
[   22.319093] amdgpu 0000:05:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 6 on hub 0
[   22.319094] amdgpu 0000:05:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 7 on hub 0
[   22.319095] amdgpu 0000:05:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 8 on hub 0
[   22.319096] amdgpu 0000:05:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 9 on hub 0
[   22.319097] amdgpu 0000:05:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 10 on hub 0
[   22.319098] amdgpu 0000:05:00.0: amdgpu: ring kiq_2.1.0 uses VM inv eng 11 on hub 0
[   22.319098] amdgpu 0000:05:00.0: amdgpu: ring sdma0 uses VM inv eng 0 on hub 1
[   22.319099] amdgpu 0000:05:00.0: amdgpu: ring vcn_dec uses VM inv eng 1 on hub 1
[   22.319100] amdgpu 0000:05:00.0: amdgpu: ring vcn_enc0 uses VM inv eng 4 on hub 1
[   22.319101] amdgpu 0000:05:00.0: amdgpu: ring vcn_enc1 uses VM inv eng 5 on hub 1
[   22.319102] amdgpu 0000:05:00.0: amdgpu: ring jpeg_dec uses VM inv eng 6 on hub 1
[   22.321130] [drm] Initialized amdgpu 3.49.0 20150101 for 0000:05:00.0 on minor 0
[   22.327038] fbcon: amdgpudrmfb (fb0) is primary device
[   23.068333] amdgpu 0000:05:00.0: [drm] fb0: amdgpudrmfb frame buffer device
```

Declaring `hardware.amdgpu.loadInInitrd = true;` results in:

```bash
$ dmesg | grep amdgpu
[    0.742672] stage-1-init: [Thu Oct 12 14:39:56 UTC 2023] loading module amdgpu...
[    2.015020] [drm] amdgpu kernel modesetting enabled.
[    2.018658] amdgpu: Virtual CRAT table created for CPU
[    2.018667] amdgpu: Topology: Add CPU node
[    2.018757] amdgpu 0000:05:00.0: enabling device (0006 -> 0007)
[    2.020627] amdgpu 0000:05:00.0: amdgpu: Fetched VBIOS from VFCT
[    2.020630] amdgpu: ATOM BIOS: 113-LUCIENNE-017
[    2.021277] amdgpu 0000:05:00.0: amdgpu: Trusted Memory Zone (TMZ) feature enabled
[    2.021282] amdgpu 0000:05:00.0: amdgpu: MODE2 reset
[    2.021362] amdgpu 0000:05:00.0: amdgpu: VRAM: 512M 0x000000F400000000 - 0x000000F41FFFFFFF (512M used)
[    2.021363] amdgpu 0000:05:00.0: amdgpu: GART: 1024M 0x0000000000000000 - 0x000000003FFFFFFF
[    2.021365] amdgpu 0000:05:00.0: amdgpu: AGP: 267419648M 0x000000F800000000 - 0x0000FFFFFFFFFFFF
[    2.021410] [drm] amdgpu: 512M of VRAM memory ready
[    2.021411] [drm] amdgpu: 15744M of GTT memory ready.
[    2.024943] amdgpu 0000:05:00.0: amdgpu: PSP runtime database doesn't exist
[    2.024945] amdgpu 0000:05:00.0: amdgpu: PSP runtime database doesn't exist
[    2.046907] amdgpu 0000:05:00.0: amdgpu: Will use PSP to load VCN firmware
[    2.831188] amdgpu 0000:05:00.0: amdgpu: RAS: optional ras ta ucode is not available
[    2.840349] amdgpu 0000:05:00.0: amdgpu: RAP: optional rap ta ucode is not available
[    2.845203] amdgpu 0000:05:00.0: amdgpu: Secure display: Generic Failure.
[    2.845212] amdgpu 0000:05:00.0: amdgpu: SECUREDISPLAY: query securedisplay TA failed. ret 0x0
[    2.846953] amdgpu 0000:05:00.0: amdgpu: SMU is initialized successfully!
[    3.018319] kfd kfd: amdgpu: Allocated 3969056 bytes on gart
[    3.018383] amdgpu: sdma_bitmap: 3
[    3.018400] amdgpu: SRAT table not found
[    3.018400] amdgpu: Virtual CRAT table created for GPU
[    3.018904] amdgpu: Topology: Add dGPU node [0x164c:0x1002]
[    3.018906] kfd kfd: amdgpu: added device 1002:164c
[    3.018980] amdgpu 0000:05:00.0: amdgpu: SE 1, SH per SE 1, CU per SH 8, active_cu_number 8
[    3.019025] amdgpu 0000:05:00.0: amdgpu: ring gfx uses VM inv eng 0 on hub 0
[    3.019027] amdgpu 0000:05:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
[    3.019027] amdgpu 0000:05:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
[    3.019028] amdgpu 0000:05:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 5 on hub 0
[    3.019029] amdgpu 0000:05:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 6 on hub 0
[    3.019030] amdgpu 0000:05:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 7 on hub 0
[    3.019031] amdgpu 0000:05:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 8 on hub 0
[    3.019032] amdgpu 0000:05:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 9 on hub 0
[    3.019032] amdgpu 0000:05:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 10 on hub 0
[    3.019033] amdgpu 0000:05:00.0: amdgpu: ring kiq_2.1.0 uses VM inv eng 11 on hub 0
[    3.019034] amdgpu 0000:05:00.0: amdgpu: ring sdma0 uses VM inv eng 0 on hub 1
[    3.019035] amdgpu 0000:05:00.0: amdgpu: ring vcn_dec uses VM inv eng 1 on hub 1
[    3.019036] amdgpu 0000:05:00.0: amdgpu: ring vcn_enc0 uses VM inv eng 4 on hub 1
[    3.019037] amdgpu 0000:05:00.0: amdgpu: ring vcn_enc1 uses VM inv eng 5 on hub 1
[    3.019038] amdgpu 0000:05:00.0: amdgpu: ring jpeg_dec uses VM inv eng 6 on hub 1
[    3.020854] [drm] Initialized amdgpu 3.49.0 20150101 for 0000:05:00.0 on minor 0
[    3.026715] fbcon: amdgpudrmfb (fb0) is primary device
[    3.760865] amdgpu 0000:05:00.0: [drm] fb0: amdgpudrmfb frame buffer device
[   16.871966] snd_hda_intel 0000:05:00.1: bound 0000:05:00.0 (ops amdgpu_dm_audio_component_bind_ops [amdgpu])
```

However, since I use LUKS disk encryption, I am prompted for my passphrase right below the following text:

```txt
<<< NixOS Stage 1 >>>

loading module amdgpu...
[    2.062381] amdgpu 0000:05:00.0; amdgpu: Secure display: Generic Failure.
[    2.062388] amdgpu 0000:05:00.0; amdgpu: SECUREDISPLAY: query securedisplay T
A failed. ret 0x0
```

This makes it impossible not to notice the `amdgpu` error, which is somewhat annoying. Are these error messages expected, or can they actually be fixed?

Reply:
You would need to have a display that supports this feature... or the kernel driver would need to be patched to no longer print these type of errors.

Reply:
`boot.plymouth.enable = true;` will also hide the output and provide a nicer password input prompt instead ;)